### ISSUE TITLE
No Asset Tag - Inventory number when motherboard doesn't recognize a Serial Number

### ISSUE THREAD

**Issue by lolokai** [fusioninventory/fusioninventory-for-glpi#2968]
Hello,
When the BIOS doesn't recognize the fields Serial Number and Inventory Number on the motherboard 

![image](https://user-images.githubusercontent.com/2960490/77883993-78f77200-7264-11ea-830b-b6992790ad4a.png)

and when we modify a record for a computer (for example we change the location) the field Inventory Number is update with the label "No Asset Tag" and the manufacturer is also update with the brand name of the Motherboard (when the brand name is recognized).

![image](https://user-images.githubusercontent.com/2960490/77883757-100ffa00-7264-11ea-8ce6-3c81523c6c50.png)

This behavior change the inventory number but we have locked this field : how to avoid this behavior ? Is that a bug ?

GLPI 9.4.5
Fusion Inventory 9.4+2.4

**Comment by lolokai** [comment]
Up ?

**Comment by kagmole** [comment]
We have the same issue in my company.

What I found:

* FusionInventory fields are locked only when they are updated; every fields are unlocked upon creation
   * `addLocks` is only called in the `item_update` GLPI hook:
     https://github.com/fusioninventory/fusioninventory-for-glpi/blob/glpi9.5/hook.php#L2000)
* As long as the FusionInventory agent on the computer is sending an empty string for the inventory number ("otherserial" in the DB), the field is not erased in GLPI, even if it is unlocked
* **However**, if a non empty string is sent by the FusionInventory agent, it will override the field _as long as_ the field is not locked
    * This means that, as long as we do not update manually at least one time the inventory number (and lock it), every time we update another field (like the location) it will be overridden

This is an issue if the FusionInventory agent send an irrelevant asset tag. I found 2 dmidecode entries with this issue:

* "No Asset Tag" with HP ProLiant
* "No Asset Information" with Lenovo ThinkPad

One way to fix it would be to make the FusionInventory agent return an empty string rather that this.

Another way would be to lock the fields upon creation.
